### PR TITLE
[core][autoscaler][v2] Instance manager impl [2/x]

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This implements an InstanceManager that updates instances to the InstanceStorage.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/v2/instance_manager/instance_manager.py [hunk 1]
```diff
@@ -315,3 +321,130 @@ def get_status_times_ns(
...
         return ts_list
+
+
+class DefaultInstanceManager(InstanceManager):
+    """
+
+    Not thread safe, should be used as a singleton.
+    """
+
+    _instance_id_counter = 0
```
Comment (scv119) [inline python/ray/autoscaler/v2/instance_manager/instance_manager.py:332]: this need to be durable in storage or uuid

File: python/ray/autoscaler/v2/instance_manager/instance_manager.py [hunk 2]
```diff
@@ -315,3 +321,130 @@ def get_status_times_ns(
             ts_list.append(status_update.timestamp_ns)
 
         return ts_list
+
+
+class DefaultInstanceManager(InstanceManager):
```
Comment (scv119) [inline python/ray/autoscaler/v2/instance_manager/instance_manager.py:326]: would we ever create another instance manager? if not we can just remove the interface

**Review by scv119 — COMMENTED** [review]
looks good, most nit

File: python/ray/autoscaler/v2/instance_manager/instance_manager.py [hunk 3]
```diff
@@ -32,24 +37,98 @@ class InstanceManager(ABC):
 
     For full status transitions, see:
     https://docs.google.com/document/d/1NzQjA8Mh-oMc-QxXOa529oneWCoA8sDiVoNkBqqDb4U/edit#heading=h.k9a1sp4qpqj4
+
+    Not thread safe, should be used as a singleton.
     """
 
-    @abstractmethod
+    _instance_id_counter = 0
```
Comment (scv119) [inline python/ray/autoscaler/v2/instance_manager/instance_manager.py:44]: add a todo?

**Review by scv119 — APPROVED** [review]

File: python/ray/autoscaler/v2/instance_manager/instance_manager.py (hunk 3 above)
Comment (rickyyx) [reply python/ray/autoscaler/v2/instance_manager/instance_manager.py:44]: oops - i should have deleted it.

**Comment by rickyyx** [comment]
- added changes w.r.t uuid for instance id.

**Pull request merged**